force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+    static void testIsAccountAddressChangedAcc2NullFalse() {
+        Account acc1 = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(acc1, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedAcc1NullFalse() {
+        Account acc2 = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(null, acc2);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedNotNullSameAddressInfoFalse() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            2,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(accounts[0], accounts[1]);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedNotNullDifferentAddressInfoTrue() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            2,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+        accounts[0].BillingStreet = 'some other billing street';
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(accounts[0], accounts[1]);
+        Test.stopTest();
+
+        System.assertEquals(true, isAccountAddressChanged);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isContactAddressEmpty *********************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void testIsContactAddressEmptyNullTrue() {
+        Test.startTest();
+        Boolean isContactAddressEmpty = ADDR_Addresses_UTIL.isContactAddressEmpty(null);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressEmpty);
+    }
+
+    @isTest
+    static void testIsContactAddressEmptyNotNullEmptyTrue() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+
+        Test.startTest();
+        Boolean isContactAddressEmpty = ADDR_Addresses_UTIL.isContactAddressEmpty(contact);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressEmpty);
+    }
+
+    @isTest
+    static void testIsContactAddressEmptyNotNullNotEmptyFalse() {
```

Comment:
Again with in an ideal world we would go field by field for this to be explicit about what is expected here.

Reply:
Agreed. maybe something we can refactor in the functional class so that we don't have to repeat the "address fields" again and again.